File: Python/Find The Original Array of Prefix Xor.py
from typing import List


class Solution:
    """
    Time:   O(n)
    Memory: O(1)
    """

    def findArray(self, pref: List[int]) -> List[int]:
        for i in reversed(range(1, len(pref))):
            pref[i] ^= pref[i - 1]
        return pref

File: Java/Find The Original Array of Prefix Xor.java
public class Solution {

    /**
     * Time:   O(n)
     * Memory: O(1)
     */
    public int[] findArray(int[] pref) {
        for (int i = pref.length - 1; i > 0; --i)
            pref[i] ^= pref[i - 1];
        return pref;
    }
}
